server/routes/courseRoutes.mjs

```diff
@@ -25,6 +25,24 @@ const router = express.Router();
  */
 router.get('/public', getAllCourses);
 
+/**
+ * @swagger
+ * /api/teacher_courses:
+ *   get:
+ *     summary: Retrieve a list of courses created by teacher (requires authentication)
+ *     tags: [Courses]
+ *     security:
+ *       - BearerAuth: []
+ *     responses:
+ *       200:
+ *         description: A list of courses.
+ *       401:
+ *         description: Unauthorized. Access token is missing or invalid.
+ *       500:
+ *         description: Internal server error.
+ */
+router.get('/teacher_courses', verifyToken, getTeacherCourses);
```

Comment:
I don't think we should have this route. We should already have a route that can get a list of courses, and we should modify that to make it work.

Maybe we could modify the `/user/:userId/course` route in `userCourseRoutes.mjs` to return the courses a user is TAKING if they are a student or the courses the user OWNS if they are a teacher?

Reply:
Maybe we can leave this as-is and refactor it in the future?

Reply:
If we're adding this new API route in this commit, why not spend a little extra time and get things right the first time?

Generally, once we release an API route, it is difficult to make big changes (like change the route name, changing the schema of the request/response, etc.) because it will break applications that are using that route. However, given that we are on a deadline and we don't have clients using our API other than the frontend, I will give this a pass this time if it's going to slow you down too much.